<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Haoyue Shuai" />


<title>R exercise: phenotype data proprocessing</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MyProject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/MoussaSang/myproject">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">R exercise: phenotype data
proprocessing</h1>
<h4 class="author">Haoyue Shuai</h4>
<h4 class="date">Oct.Â 8th, 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-10-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>myproject/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, youâll want to first commit it to
the Git repo. If youâre still working on the analysis, you can ignore
this warning. When youâre finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity itâs best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230926code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230926)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230926code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230926)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMoussaSangmyprojecttreeab6caff30993d0df9108204787fd7e115924c193targetblankab6caffa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/MoussaSang/myproject/tree/ab6caff30993d0df9108204787fd7e115924c193" target="_blank">ab6caff</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomMoussaSangmyprojecttreeab6caff30993d0df9108204787fd7e115924c193targetblankab6caffa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/MoussaSang/myproject/tree/ab6caff30993d0df9108204787fd7e115924c193" target="_blank">ab6caff</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .sos/

Untracked files:
    Untracked:  analysis/data_cleaned.csv
    Untracked:  analysis/orientation.Rmd
    Untracked:  config.default.yml
    Untracked:  config.yml
    Untracked:  data/data_cleaned.csv
    Untracked:  data_cleaned.csv
    Untracked:  jnbinder_docker.sh
    Untracked:  notebook/
    Untracked:  workflow/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g.Â HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div style="page-break-after: always;"></div>
<div id="instruction" class="section level2">
<h2>Instruction</h2>
<p>This exercise was adapted by Haoyue Shuai from one of her analysis on
phenotype data in preparation for genetic association studies. Donât
panic if youâre just starting to learn R. The exercise does not involve
writing serious software programs in R, but rather to execute some
interactive and intuitive commands to get some preliminary phenotype
data analysis done. There will be some questions throughout that you can
answer using your own words (and some data science intuition) with a few
sentences. Some of them involve writing additional codes but you can
mostly find and modify codes we provide as examples to address the
questions. You do not have to answer all the questions but we strongly
encourage you to make an attempt.</p>
<p>Please save your work as you complete them, but <strong>do
not</strong> push your version of the file back to your forked github
repo. Instead please follow the instructions in Task 4 to report the
analysis back to us.</p>
<div id="dataset" class="section level3">
<h3>Dataset</h3>
<p>The data-set can be found as
<code>data/UKB_Phenotype/data_cleaned.csv</code>.</p>
<div style="page-break-after: always;"></div>
<div style="page-break-after: always;"></div>
</div>
</div>
<div id="genetic-association-study" class="section level2">
<h2>Genetic association study</h2>
<p>We perform genetic association studies to identify genetic factors
(variants) that may be involved in a complex trait (tinnitus, asthma,
etc) etiology. In brief, genetic association studies compare and
identify difference in genetic data of individuals with disease (cases)
to those without (controls). We report genetic variants that are
observed more frequently in cases than in controls.</p>
<p>In order to perform genetic association studies, we need phenotype
data and genotype data from individuals we collect.</p>
<ul>
<li>Phenotype and covariate data: age, sex, height, weight, condition
for that trait (tinnitus in the example below case), etc.</li>
<li>Genotype data: You can roughly understand it as a sequence of the
bases in DNA molecules, A/T/C/G, for all chromosomes in human
genome.</li>
</ul>
<div id="disease-phenotype-data" class="section level3">
<h3>Disease phenotype data</h3>
<p>We use a toy data-set extracted from the UK Biobank project.</p>
<p><strong>Load the data</strong></p>
<p>Load <code>UKB_Phenotype/data_cleaned.csv</code>. Note, you only need
to complete the codes when seeing <code>YOUR CODE</code>. Please execute
other existing codes as is.</p>
<pre class="r"><code># you need to put the data-set in the same folder
# where this .rmd file sits,
# which is here:

getwd()
sub_UKBB&lt;-read.csv(&quot;data_cleaned.csv&quot;)</code></pre>
<pre><code>[1] &quot;/home/h3africa/myproject&quot;</code></pre>
<p><strong>Exploratory data analysis (EDA) of the data</strong></p>
<pre class="r"><code>dim(sub_UKBB) # This data covering 144756 participants and 11 variables of them (IID, FID, etc)</code></pre>
<pre><code>[1] 144756     11</code></pre>
<pre class="r"><code>colnames(sub_UKBB) # 11 variables </code></pre>
<pre><code> [1] &quot;IID&quot;         &quot;FID&quot;         &quot;f.4803.0.0&quot;  &quot;f.4803.1.0&quot;  &quot;f.4803.2.0&quot; 
 [6] &quot;f.4803.3.0&quot;  &quot;f.22001.0.0&quot; &quot;f.21003.0.0&quot; &quot;f.21003.1.0&quot; &quot;f.21003.2.0&quot;
[11] &quot;f.21003.3.0&quot;</code></pre>
<pre class="r"><code>summary(sub_UKBB) </code></pre>
<pre><code>      IID               FID           f.4803.0.0         f.4803.1.0       
 Min.   :1000046   Min.   :1000046   Length:144756      Length:144756     
 1st Qu.:2267981   1st Qu.:2267981   Class :character   Class :character  
 Median :3527260   Median :3527260   Mode  :character   Mode  :character  
 Mean   :3521488   Mean   :3521488                                        
 3rd Qu.:4779335   3rd Qu.:4779335                                        
 Max.   :6025411   Max.   :6025411                                        
                                                                          
  f.4803.2.0         f.4803.3.0        f.22001.0.0         f.21003.0.0   
 Length:144756      Length:144756      Length:144756      Min.   :40.00  
 Class :character   Class :character   Class :character   1st Qu.:51.00  
 Mode  :character   Mode  :character   Mode  :character   Median :58.00  
                                                          Mean   :56.98  
                                                          3rd Qu.:63.00  
                                                          Max.   :73.00  
                                                                         
  f.21003.1.0      f.21003.2.0      f.21003.3.0    
 Min.   :43.00    Min.   :45.00    Min.   :50.00   
 1st Qu.:57.00    1st Qu.:58.00    1st Qu.:59.00   
 Median :63.00    Median :65.00    Median :66.00   
 Mean   :61.69    Mean   :64.36    Mean   :65.12   
 3rd Qu.:67.00    3rd Qu.:70.00    3rd Qu.:71.00   
 Max.   :79.00    Max.   :82.00    Max.   :82.00   
 NA&#39;s   :130131   NA&#39;s   :110381   NA&#39;s   :142607  </code></pre>
<pre class="r"><code>head(sub_UKBB) # show part of the data</code></pre>
<pre><code>      IID     FID                             f.4803.0.0 f.4803.1.0 f.4803.2.0
1 1000046 1000046                                   &lt;NA&gt;       &lt;NA&gt;  No, never
2 1000063 1000063                              No, never       &lt;NA&gt;       &lt;NA&gt;
3 1000078 1000078                                   &lt;NA&gt;  No, never  No, never
4 1000105 1000105 Yes, but not now, but have in the past       &lt;NA&gt;       &lt;NA&gt;
5 1000112 1000112                                   &lt;NA&gt;       &lt;NA&gt;  No, never
6 1000141 1000141                              No, never       &lt;NA&gt;       &lt;NA&gt;
  f.4803.3.0 f.22001.0.0 f.21003.0.0 f.21003.1.0 f.21003.2.0 f.21003.3.0
1       &lt;NA&gt;      Female          62          NA          73          NA
2       &lt;NA&gt;        Male          43          NA          NA          NA
3       &lt;NA&gt;      Female          52          57          60          NA
4       &lt;NA&gt;      Female          54          NA          NA          NA
5       &lt;NA&gt;        Male          58          NA          68          NA
6       &lt;NA&gt;      Female          49          NA          NA          NA</code></pre>
<p><strong>Q1:</strong> How many females and males are there in this
data? Please show your code below how these numbers are computed.</p>
<p><strong>There are 77535 females and 67221 males in this
data.</strong></p>
<pre class="r"><code>table(sub_UKBB$f.22001.0.0 == &quot;Female&quot;)</code></pre>
<pre><code>
FALSE  TRUE 
67221 77535 </code></pre>
<p><strong>Q2:</strong> What type of trait do you think best describes
tinnitus as a phenotype? A.Binary B. Continuous C. Ordinal</p>
<p><strong>The Binary trait best describes the tinnitus as a
phenotype.</strong></p>
<p><strong>Q3:</strong> Recode f.4803</p>
<p>Field 4803 (f.4803) is the answers from participants for ACE
touchscreen question âDo you get or have you had noises (such as ringing
or buzzing) in your head or in one or both ears that lasts for more than
five minutes at a time?â</p>
<p>These fields contains answers to the questions in their first, 2nd,
3rd and 4th hospital visit: f.4803.0.0, f.4803.1.0, f.4803.2.0,
f.4803.3.0.</p>
<pre class="r"><code># Recode function:
recode&lt;-function(df,column_name){
  new_names&lt;-c()
  for (i in column_name){
    new_column_name&lt;-paste0(i,&quot;_recode&quot;)
    new_names&lt;-c(new_names,new_column_name)
    df[,new_column_name] &lt;- revalue(df[,i], c(&quot;No, never&quot;= 0, 
                                            &quot;Yes, but not now, but have in the past&quot;= 1, 
                                            &quot;Yes, now some of the time&quot;= 1, 
                                            &quot;Yes, now a lot of the time&quot;= 1,
                                            &quot;Yes, now most or all of the time&quot;= 1,
                                            &quot;Prefer not to answer&quot;= NA,
                                            &quot;Do not know&quot;= NA ))
  }
  return (list(df=df,new_column_names=new_names))
}

# columns needs to be recoded:
column_name&lt;-c(&quot;f.4803.0.0&quot;,&quot;f.4803.1.0&quot;,&quot;f.4803.2.0&quot;,&quot;f.4803.3.0&quot;)

# get a new data.frame with recoded columns added:
df_recode&lt;-recode(df=sub_UKBB,column_name)$df

# get names of recoded columns:
new_column_names&lt;-recode(df=sub_UKBB,column_name)$new_column_names

# show recode summary:
for (i in new_column_names)
{cat(i,&quot;summary:&quot;);print(table(df_recode[,i]));cat(&quot;\n&quot;)}</code></pre>
<pre><code>f.4803.0.0_recode summary:
    0     1 
80805 31685 

f.4803.1.0_recode summary:
    0     1 
10282  4214 

f.4803.2.0_recode summary:
    0     1 
22583 11405 

f.4803.3.0_recode summary:
   0    1 
1451  691 </code></pre>
<p>What do you think has been achieved by recoding these fields?</p>
<p><strong>The variables have been turned into three categories: 0, 1
and NA</strong></p>
<p><strong>Q4:</strong> Define case and control status of tinnitus for
each participant in the study:</p>
<pre class="r"><code>data_sub &lt;- df_recode[,new_column_names]
# Function to define cases
f&lt;-function(x){
  visit&lt;-c()
  for (i in 1:4){
    if (!is.na(x[i]))
    {visit&lt;-c(visit,x[i])}
  }
  if (&quot;1&quot; %in% visit){result= TRUE}
  else{result=FALSE}
  return (result)
}
# Apply the above function
df_recode$cases&lt;-apply(data_sub, 1, f)
head(df_recode,10)</code></pre>
<pre><code>       IID     FID                             f.4803.0.0 f.4803.1.0 f.4803.2.0
1  1000046 1000046                                   &lt;NA&gt;       &lt;NA&gt;  No, never
2  1000063 1000063                              No, never       &lt;NA&gt;       &lt;NA&gt;
3  1000078 1000078                                   &lt;NA&gt;  No, never  No, never
4  1000105 1000105 Yes, but not now, but have in the past       &lt;NA&gt;       &lt;NA&gt;
5  1000112 1000112                                   &lt;NA&gt;       &lt;NA&gt;  No, never
6  1000141 1000141                              No, never       &lt;NA&gt;       &lt;NA&gt;
7  1000236 1000236                              No, never       &lt;NA&gt;  No, never
8  1000248 1000248                              No, never       &lt;NA&gt;       &lt;NA&gt;
9  1000269 1000269                                   &lt;NA&gt;       &lt;NA&gt;  No, never
10 1000340 1000340                              No, never       &lt;NA&gt;       &lt;NA&gt;
   f.4803.3.0 f.22001.0.0 f.21003.0.0 f.21003.1.0 f.21003.2.0 f.21003.3.0
1        &lt;NA&gt;      Female          62          NA          73          NA
2        &lt;NA&gt;        Male          43          NA          NA          NA
3        &lt;NA&gt;      Female          52          57          60          NA
4        &lt;NA&gt;      Female          54          NA          NA          NA
5        &lt;NA&gt;        Male          58          NA          68          NA
6        &lt;NA&gt;      Female          49          NA          NA          NA
7        &lt;NA&gt;        Male          61          NA          70          NA
8        &lt;NA&gt;        Male          63          NA          NA          NA
9        &lt;NA&gt;        Male          60          NA          71          NA
10       &lt;NA&gt;      Female          54          NA          NA          NA
   f.4803.0.0_recode f.4803.1.0_recode f.4803.2.0_recode f.4803.3.0_recode
1               &lt;NA&gt;              &lt;NA&gt;                 0              &lt;NA&gt;
2                  0              &lt;NA&gt;              &lt;NA&gt;              &lt;NA&gt;
3               &lt;NA&gt;                 0                 0              &lt;NA&gt;
4                  1              &lt;NA&gt;              &lt;NA&gt;              &lt;NA&gt;
5               &lt;NA&gt;              &lt;NA&gt;                 0              &lt;NA&gt;
6                  0              &lt;NA&gt;              &lt;NA&gt;              &lt;NA&gt;
7                  0              &lt;NA&gt;                 0              &lt;NA&gt;
8                  0              &lt;NA&gt;              &lt;NA&gt;              &lt;NA&gt;
9               &lt;NA&gt;              &lt;NA&gt;                 0              &lt;NA&gt;
10                 0              &lt;NA&gt;              &lt;NA&gt;              &lt;NA&gt;
   cases
1  FALSE
2  FALSE
3  FALSE
4   TRUE
5  FALSE
6  FALSE
7  FALSE
8  FALSE
9  FALSE
10 FALSE</code></pre>
<p>How many cases and how many controls do we have for this
phenotpype?</p>
<p>table(df_recode[,16])</p>
<p><strong><em>For this phenotype, we have 43206 cases and 101550
controls.</em></strong></p>
<p><strong>Q5:</strong> Extract a subset of columns from all
participants for association study.</p>
<pre class="r"><code>df_cases  &lt;- df_recode %&gt;%
  select(IID,FID,cases)%&gt;% 
  filter(cases==TRUE)
head(df_cases,10)  </code></pre>
<pre><code>       IID     FID cases
1  1000105 1000105  TRUE
2  1000357 1000357  TRUE
3  1000443 1000443  TRUE
4  1000494 1000494  TRUE
5  1000625 1000625  TRUE
6  1000713 1000713  TRUE
7  1000728 1000728  TRUE
8  1000745 1000745  TRUE
9  1000875 1000875  TRUE
10 1000992 1000992  TRUE</code></pre>
<p>Please modify codes above to extract all the controls and keep only
these columns: <code>FID</code>, <code>IID</code>, <code>cases</code>,
<code>f.22001.0.0</code>, <code>f.21003.0.0</code>,
<code>f.21003.1.0</code>, <code>f.21003.2.0</code>,
<code>f.21003.3.0</code>. Please show the first 10 rows of the
output.</p>
<pre class="r"><code># YOUR CODE
df_cases  &lt;- df_recode %&gt;%
  select(IID,FID,cases, `f.22001.0.0`, `f.21003.0.0`, `f.21003.1.0`, `f.21003.2.0`, `f.21003.3.0`)%&gt;% 
  filter(cases==TRUE)
head(df_cases,10)  </code></pre>
<pre><code>       IID     FID cases f.22001.0.0 f.21003.0.0 f.21003.1.0 f.21003.2.0
1  1000105 1000105  TRUE      Female          54          NA          NA
2  1000357 1000357  TRUE      Female          69          NA          NA
3  1000443 1000443  TRUE      Female          45          NA          52
4  1000494 1000494  TRUE        Male          61          NA          NA
5  1000625 1000625  TRUE      Female          60          NA          NA
6  1000713 1000713  TRUE        Male          58          NA          67
7  1000728 1000728  TRUE        Male          61          NA          NA
8  1000745 1000745  TRUE        Male          60          NA          NA
9  1000875 1000875  TRUE        Male          59          NA          NA
10 1000992 1000992  TRUE        Male          48          NA          56
   f.21003.3.0
1           NA
2           NA
3           NA
4           NA
5           NA
6           NA
7           NA
8           NA
9           NA
10          NA</code></pre>
<div style="page-break-after: always;"></div>
</div>
<div id="covariates" class="section level3">
<h3>Covariates</h3>
<p><strong>Q6:</strong> Field 21003 contains the information of the age
of participants, same as field 4803. Note that some of them have more
than one age. Can you guess why?</p>
<p><strong>The fact of having more than one age can be explained by the
different ages at the first, 2nd, 3rd and so on hospital
visit.</strong></p>
<p><strong>Q7:</strong> For those with more than one age records, which
age do you think should be used in the genetic association analysis?</p>
<p><strong>The median age is the one which should be used for the
genetic association study.</strong></p>
<p><strong>Q8:</strong> Please compute a summary of age information for
controls (you can use <code>summary()</code> function in R):</p>
<pre class="r"><code>#YOUR CODE
df_controls  &lt;- df_recode %&gt;%
  select(IID,FID,cases, `f.22001.0.0`, `f.21003.0.0`, `f.21003.1.0`, `f.21003.2.0`, `f.21003.3.0`)%&gt;% 
  filter(cases==FALSE)
head(df_controls, 10)
summary(df_controls$f.21003.0.0)</code></pre>
<pre><code>       IID     FID cases f.22001.0.0 f.21003.0.0 f.21003.1.0 f.21003.2.0
1  1000046 1000046 FALSE      Female          62          NA          73
2  1000063 1000063 FALSE        Male          43          NA          NA
3  1000078 1000078 FALSE      Female          52          57          60
4  1000112 1000112 FALSE        Male          58          NA          68
5  1000141 1000141 FALSE      Female          49          NA          NA
6  1000236 1000236 FALSE        Male          61          NA          70
7  1000248 1000248 FALSE        Male          63          NA          NA
8  1000269 1000269 FALSE        Male          60          NA          71
9  1000340 1000340 FALSE      Female          54          NA          NA
10 1000379 1000379 FALSE      Female          51          NA          NA
   f.21003.3.0
1           NA
2           NA
3           NA
4           NA
5           NA
6           NA
7           NA
8           NA
9           NA
10          NA
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  40.00   51.00   58.00   56.76   63.00   73.00 </code></pre>
<div style="page-break-after: always;"></div>
</div>
<div id="association-testing-via-regression-analysis"
class="section level3">
<h3>Association testing via regression analysis</h3>
<p>To identify genetic factors that may be involved in this trait
(tinnitus), we would need to find the association between the genotype
and the phenotype. Regression analysis is the basis of many association
analysis. Instead of overwhelming you with huge genotype data, we use
here a simple dataset for regression analysis to demonstrate what
association studies look like.</p>
<p>We fit below simple linear model with 2 variables from a data-set to
see their relationship. For example <code>mpg</code>
vs.Â <code>weight</code> in this <code>ISLR::Auto</code> data-set.</p>
<p><strong>Q9:</strong> Is there association between
<code>mpg of the car</code> and <code>weight of the car</code>? If so,
it appearing to be positive or negative? Is the association significant
and why?</p>
<p><strong>There is an association between <code>mpg of the car</code>
and <code>its weight</code>.</strong> <strong>This appears to be a
positive correlation</strong> <strong>The association is significant,
because the p-value is less than 5%</strong></p>
<pre><code>Loading required namespace: ISLR</code></pre>
<pre><code>
Call:
lm(formula = mpg ~ weight, data = auto_data)

Residuals:
     Min       1Q   Median       3Q      Max 
-11.9736  -2.7556  -0.3358   2.1379  16.5194 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 46.216524   0.798673   57.87   &lt;2e-16 ***
weight      -0.007647   0.000258  -29.64   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 4.333 on 390 degrees of freedom
Multiple R-squared:  0.6926,    Adjusted R-squared:  0.6918 
F-statistic: 878.8 on 1 and 390 DF,  p-value: &lt; 2.2e-16</code></pre>
<p><strong>Q10:</strong> Please create a new variable to indicate cars
having MPG greater than 23 as 1, otherwise 0, then use logistic
regression via <code>glm()</code> function to analyze association
between weight and this new variable you just created. Please comment on
what you find.</p>
<p><strong>Q11:</strong> Find the <code>Estimates</code> from your
association results summary. How do you interpret the estimated effects
of weight in the context of linear regression, and in the context of
logistic regression?</p>
<pre class="r"><code>#YOUR CODE</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 Patched (2022-11-10 r83330)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 23.04

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.21.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] reshape2_1.4.4  pROC_1.18.4     knitr_1.44      rapportools_1.1
 [5] ggpubr_0.6.0    pander_0.6.5    forcats_1.0.0   stringr_1.5.0  
 [9] purrr_1.0.2     readr_2.1.4     tidyr_1.3.0     tibble_3.2.1   
[13] tidyverse_1.3.2 plyr_1.8.9      ggplot2_3.4.3   dplyr_1.1.3    
[17] pacman_0.5.1   

loaded via a namespace (and not attached):
 [1] httr_1.4.4          sass_0.4.7          jsonlite_1.8.7     
 [4] carData_3.0-5       modelr_0.1.11       bslib_0.5.1        
 [7] googlesheets4_1.0.1 cellranger_1.1.0    yaml_2.3.7         
[10] pillar_1.9.0        backports_1.4.1     glue_1.6.2         
[13] digest_0.6.33       promises_1.2.0.1    ggsignif_0.6.4     
[16] rvest_1.0.3         colorspace_2.1-0    htmltools_0.5.6    
[19] httpuv_1.6.9        pkgconfig_2.0.3     broom_1.0.5        
[22] haven_2.5.3         scales_1.2.1        later_1.3.0        
[25] tzdb_0.4.0          timechange_0.2.0    git2r_0.32.0       
[28] googledrive_2.0.0   generics_0.1.3      car_3.1-2          
[31] cachem_1.0.8        withr_2.5.1         cli_3.6.1          
[34] magrittr_2.0.3      crayon_1.5.2        readxl_1.4.3       
[37] evaluate_0.22       fs_1.6.3            fansi_1.0.4        
[40] MASS_7.3-58.2       rstatix_0.7.2       xml2_1.3.3         
[43] tools_4.2.2         hms_1.1.3           gargle_1.3.0       
[46] lifecycle_1.0.3     munsell_0.5.0       reprex_2.0.2       
[49] compiler_4.2.2      jquerylib_0.1.4     rlang_1.1.1        
[52] grid_4.2.2          rstudioapi_0.15.0   ISLR_1.4           
[55] rmarkdown_2.25      gtable_0.3.4        abind_1.4-5        
[58] DBI_1.1.3           R6_2.5.1            lubridate_1.9.3    
[61] fastmap_1.1.1       utf8_1.2.3          workflowr_1.7.1    
[64] rprojroot_2.0.3     stringi_1.7.12      Rcpp_1.0.10        
[67] vctrs_0.6.3         dbplyr_2.3.4        tidyselect_1.2.0   
[70] xfun_0.40          </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
